pkg/querier/codec/protobuf_codec.go

```diff
@@ -129,18 +153,63 @@ func getVectorSamples(data *v1.QueryData) *[]tripperware.Sample {
 				}
 			}
 		}
-
-		vectorSamples[i] = tripperware.Sample{
-			Labels: labels,
-			Sample: &cortexpb.Sample{
+		vectorSamples[i].Labels = labels
+		
+		if data.Result.(promql.Vector)[i].H != nil {
```

Comment:
In the `json_codec` implementation we cannot have both a histogram and float at the same time for a vector sample